lib/src/xdg_screen_cast_portal.dart

```diff
@@ -0,0 +1,254 @@
+  final List<int> size;
+
+  /// The type of the content which is being screen casted.
+  final ScreenCastAvailableSourceType sourceType;
+
+  ScreenCastStream({
+    required this.nodeId,
+    required this.id,
+    required this.position,
+    required this.size,
+    required this.sourceType,
+  });
+  @override
+  int get hashCode => Object.hash(
+      nodeId, id, Object.hashAll(position), Object.hashAll(size), sourceType);
+
+  @override
+  bool operator ==(other) {
+    if (identical(this, other)) return true;
+    final listEquals = const DeepCollectionEquality().equals;
+    return other is ScreenCastStream &&
+        other.nodeId == nodeId &&
+        other.id == id &&
+        listEquals(other.position, position) &&
+        listEquals(other.size, size) &&
+        other.sourceType == sourceType;
+  }
+
+  @override
+  String toString() =>
+      '$runtimeType(node ID: $nodeId, id: $id, position: $position, size: $size, sourceType: $sourceType})';
+}
+
+/// Screen cast portal.
+class XdgScreenCastPortal {
+  final DBusRemoteObject _object;
+  final String Function() _generateToken;
+
+  DBusObjectPath? _sessionPath;
+
+  XdgScreenCastPortal(this._object, this._generateToken);
+
+  /// Get the version of this portal.
+  Future<int> getVersion() => _object
+      .getProperty('org.freedesktop.portal.ScreenCast', 'version',
+          signature: DBusSignature('u'))
+      .then((v) => v.asUint32());
+
+  /// Get the available screen cast source types.
+  Future<Set<ScreenCastAvailableSourceType>> getAvailableSourceTypes() =>
+      _object
+          .getProperty(
+              'org.freedesktop.portal.ScreenCast', 'AvailableSourceTypes',
+              signature: DBusSignature('u'))
+          .then((v) {
+        final value = v.asUint32();
+        final types = <ScreenCastAvailableSourceType>{};
+        if (value & 1 != 0) types.add(ScreenCastAvailableSourceType.monitor);
+        if (value & 2 != 0) types.add(ScreenCastAvailableSourceType.window);
+        if (value & 4 != 0) types.add(ScreenCastAvailableSourceType.virtual);
+        return types;
+      });
+
+  /// Get the available screen cast cursor modes.
+  Future<Set<ScreenCastAvailableCursorMode>> getAvailableCursorModes() =>
+      _object
+          .getProperty(
+              'org.freedesktop.portal.ScreenCast', 'AvailableCursorModes',
+              signature: DBusSignature('u'))
+          .then((v) {
+        final value = v.asUint32();
+        final types = <ScreenCastAvailableCursorMode>{};
+        if (value & 1 != 0) types.add(ScreenCastAvailableCursorMode.hidden);
+        if (value & 2 != 0) types.add(ScreenCastAvailableCursorMode.embedded);
+        if (value & 4 != 0) types.add(ScreenCastAvailableCursorMode.metadata);
+        return types;
+      });
+
+  /// Create a location session that returns a stream of location updates from the portal.
+  /// When the session is no longer required close the stream.
+  Future<void> createSession() async {
```

Comment:
`selectSources` and `start` should be merged into this method, and this will fix the issue of the shared `_sessionPath` object. See the location portal for how this is done.

Reply:
I merged selectSources and start. But method openPipeWireRemote depend of  session path.